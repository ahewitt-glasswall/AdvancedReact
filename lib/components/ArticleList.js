import React, { memo } from "react";

import Article from "./Article";

const ArticleList = ({ articles }) => {
    return (
        <div>
            {Object.values(articles).map(article =>
                <Article key={article.id} article={article} />)}
        </div>
    );
};

export default memo(ArticleList);