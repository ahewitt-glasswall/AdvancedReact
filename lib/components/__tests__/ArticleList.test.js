import React from "react";

import ArticleList from "../ArticleList";

import { shallow } from "enzyme";

describe("ArticleList", () => {

    const testProps = {
        articles: {
            a: { id: "articleA", date: "01/01/01", title: "Article A", body: "body" },
            b: { id: "articleB", date: "01/01/01", title: "Article B", body: "body" }
        }
    };

    it("renders correctly", () => {
        const wrapper = shallow(
            <ArticleList {...testProps} />);

        expect(wrapper.find("ArticleContainer").length).toBe(2);

        expect(wrapper).toMatchSnapshot();
    });
});