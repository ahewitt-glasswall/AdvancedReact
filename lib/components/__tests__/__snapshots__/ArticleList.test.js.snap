// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <ArticleContainer
        article={
            Object {
                "body": "body",
                "date": "01/01/01",
                "id": "articleA",
                "title": "Article A",
              }
        }
    />
    <ArticleContainer
        article={
            Object {
                "body": "body",
                "date": "01/01/01",
                "id": "articleB",
                "title": "Article B",
              }
        }
    />
</div>,
  "nodes": Array [
    <div>
      <ArticleContainer
            article={
                  Object {
                        "body": "body",
                        "date": "01/01/01",
                        "id": "articleA",
                        "title": "Article A",
                      }
            }
      />
      <ArticleContainer
            article={
                  Object {
                        "body": "body",
                        "date": "01/01/01",
                        "id": "articleB",
                        "title": "Article B",
                      }
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useDeferredValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useResponder": [Function],
      "useState": [Function],
      "useTransition": [Function],
    },
    "_element": <ArticleList
      articles={
            Object {
                  "a": Object {
                    "body": "body",
                    "date": "01/01/01",
                    "id": "articleA",
                    "title": "Article A",
                  },
                  "b": Object {
                    "body": "body",
                    "date": "01/01/01",
                    "id": "articleB",
                    "title": "Article B",
                  },
                }
      }
/>,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_instance": null,
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <div>
      <ArticleContainer
            article={
                  Object {
                        "body": "body",
                        "date": "01/01/01",
                        "id": "articleA",
                        "title": "Article A",
                      }
            }
      />
      <ArticleContainer
            article={
                  Object {
                        "body": "body",
                        "date": "01/01/01",
                        "id": "articleB",
                        "title": "Article B",
                      }
            }
      />
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ArticleList
    articles={
        Object {
            "a": Object {
              "body": "body",
              "date": "01/01/01",
              "id": "articleA",
              "title": "Article A",
            },
            "b": Object {
              "body": "body",
              "date": "01/01/01",
              "id": "articleB",
              "title": "Article B",
            },
          }
    }
/>,
}
`;
